clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/util/UpsertIdGenerator.scala

```diff
@@ -0,0 +1,96 @@
+package org.broadinstitute.clio.server.dataaccess.util
+
+import scala.util.Random
+
+/**
+  * Generate a lexically-ordered, unique ID based on the Firebase Push ID.
+  *
+  * A Push ID is a 20-byte string of by 8 characters of timestamp
+  * followed by 12 characters of randomized (yet sequential) data.
+  *
+  * A recent one looks like this: -KuzbQJIFBhwvtkvrBHF
+  *
+  * @see [[https://firebase.googleblog.com/2015/02/the-2120-ways-to-ensure-unique_68.html]]
+  *
+  */
+object UpsertIdGenerator {
+
+  private val push =
+    "-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz"
+
+  private val source = new Random()
+
+  private val timestampCount = 8
+  private val randomCount = 12
```

Comment:
I think Scala conventions for static constants is to use upper camel case, e.g. `RandomCount`. Also it would be nice to have comments for the vals here.

Reply:
Their names are their comments.
Virtually everything is constant the way we write Scala here, but OK.
The object comment explains those 2 constants.